helix-core/src/main/java/org/apache/helix/tools/ClusterVerifiers/StrictMatchExternalViewVerifier.java

```diff
@@ -279,17 +303,21 @@ private boolean verifyExternalView(ResourceControllerDataProvider dataCache, Ext
     String stateModelDefName = idealState.getStateModelDefRef();
     StateModelDefinition stateModelDef = cache.getStateModelDef(stateModelDefName);
 
-    Map<String, Map<String, String>> idealPartitionState =
-        new HashMap<String, Map<String, String>>();
-
-    Set<String> liveEnabledInstances = new HashSet<String>(cache.getLiveInstances().keySet());
-    liveEnabledInstances.removeAll(cache.getDisabledInstances());
+    Map<String, Map<String, String>> idealPartitionState = new HashMap<>();
 
     for (String partition : idealState.getPartitionSet()) {
       List<String> preferenceList = AbstractRebalancer
-          .getPreferenceList(new Partition(partition), idealState, liveEnabledInstances);
-      Map<String, String> idealMapping =
-          HelixUtil.computeIdealMapping(preferenceList, stateModelDef, liveEnabledInstances);
+          .getPreferenceList(new Partition(partition), idealState, cache.getEnabledLiveInstances());
+      Map<String, String> idealMapping;
+      if (_isDeactivatedNodeAware) {
+        idealMapping = HelixUtil
```

Comment:
This was a bug in the Util function. But it has been fixed in https://github.com/apache/helix/commit/f6f50c2ab72131be6072a74ec558a9147e979687. Will this change for _isDeactivatedNodeAware necessary.

Reply:
The strict ev verifier uses a different method. f6f50c2 won't impact the verifier's behavior. That's why the tests cannot pass for WAGED rebalancer even after I merged the master.